### ISSUE TITLE
Cookie based login will be deprecated soon. Does this library support OAuth?

### ISSUE THREAD

**Issue by clearitout** [samsymons/RedditKit#61]
https://www.reddit.com/r/redditdev/comments/2ujhkr/important_api_licensing_terms_clarified/

**Comment by samsymons** [comment]
It doesn't, but it's a work in progress. I hope to have something to try out in a couple weeks.

**Comment by pkamb** [comment]
OAuth branch:  
https://github.com/samsymons/RedditKit/tree/oauth

**Comment by samsymons** [comment]
I'm hoping to have OAuth publicly available some time this coming week – the current branch is very out of date unfortunately, but I've been working on it all day today and it's looking pretty good.

**Comment by ealimam** [comment]
Sounds great, can't wait! Thanks Sam!

**Comment by samsymons** [comment]
@ealimam No problem, sorry for taking so long with it!

**Comment by ealimam** [comment]
@samsymons It's gonna be worth it, I'm sure!

**Comment by WordMasterRice** [comment]
Any update on OAuth support?  Would love to test this out.

**Comment by ealimam** [comment]
Hi @samsymons,

Do you have any updates on OAuth support? Would love to get this in as the [August 3rd deadline](https://www.reddit.com/r/redditdev/comments/2ujhkr/important_api_licensing_terms_clarified/) is right around the corner. :)

Thanks!
Emory

**Comment by ealimam** [comment]
Hi @samsymons Any word on this?

**Comment by samsymons** [comment]
@ealimam Hey Emory – things are looking pretty good at the moment. Later this coming week (probably the 25th) there'll be the first beta of 2.0 available, with the plan to release it very soon after, so that we can get ahead of the August 3 deadline.

Sorry again for it taking so long; I'm currently one of two developers at a company so spare time is pretty limited. :)

**Comment by ealimam** [comment]
@samsymons Yes, I certainly know how it goes. ;) Sounds great, can't wait to see all the good stuff you've been working on!

**Comment by pkamb** [comment]
Hi @samsymons, please let me know if there's I can do to help with OAuth support.

I'm suddenly realizing that [Reddit Notifier](https://itunes.apple.com/us/app/reddit-notifier/id468366517?mt=12) will need an update quickly or I'll have many unhappy customers on 8/3!

If you had a WIP branch or checklist of remaining items I would be happy to help. Let me know!

**Comment by samsymons** [comment]
Hey @pkamb – there's a [beta version of 2.0 with OAuth support](https://github.com/samsymons/RedditKit/tree/2.0-beta1) currently available, but the API is in flux and a lot of the new features aren't there just yet, like live thread support.

I'm debating releasing 2.0 once the OAuth API is finalised, and then add the newer endpoints in future releases, just to get it available sooner. I've recently found myself with a lot more free time, so I'll be on RedditKit more or less full time until it's out. :)

**Comment by pkamb** [comment]
Great, thanks for that. Hoping to put some time into this and my app soon.

FYI, everyone's going to have a bit more time:

https://www.reddit.com/comments/37e2mv/

> The biggest project that affects you directly is the oauth transition that we had planned for August. We understand that forcing this move without adequate support from our side is not fair to the dev community so until we have time to help ease this transition we will not be forcing the swap. Note that this is still something we will be pursuing and any new features we release will continue to be supported only on oauth.

**Comment by pavementscratcher** [comment]
Curious, is this going to be added any time soon? This is a really important feature and I haven't heard anything in ages.
